### PR TITLE
Lazy wheel

### PR DESCRIPTION
**Pull request by nmerriam**
## Description
This is a wheel algorithm but without a statically pre-computed prime set. Rather, the wheel is computed dynamically (lazily) after a concept by Colin Runciman, see https://eprints.whiterose.ac.uk/3784/1/runcimanc1.pdf

## Contributing requirements
As far as I can tell, this algorithm is significantly different from any current implementation.

* [x] I read the contribution guidelines in CONTRIBUTING.md.
* [x] I placed my solution in the correct solution folder.
* [x] I added a README.md with the right badge(s).
* [x] I added a Dockerfile that builds and runs my solution.
* [x] I selected `drag-race` as the target branch.
* [x] All code herein is licensed compatible with BSD-3.

### CODE REVIEW

File: PrimeC/solution_4/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM ubuntu
```
Comment (rbergen) [inline PrimeC/solution_4/Dockerfile]: Please pin the base image down to a specific version.

**Comment by nmerriam** [comment]
Hi Rutger,

Thank you for getting in touch. I am afraid I have never done anything with
Docker or GitHub before and I don't know what you mean. I got a friend to
coach me through the steps as far as here and I don't really understand
what's going on.

I have put Hans on cc. Maybe you can straighten this out?

Sorry to be so helpless. One of my motivations for getting involved was to
learn more.

Best wishes
Nick

**Comment by rbergen** [comment]
@nmerriam That's ok, it's one of the learning curves everybody needs to go through.

Your Dockerfile indicates it wants to use an Ubuntu docker image as a base image with this directive:
```dockerfile
FROM ubuntu
```

What's needed is that you make explicit what version of Ubuntu base image you want to use. We prefer you use a Long Time Support image like 20.04, which would be specified like this:
```dockerfile
FROM ubuntu:20.04
```

If there is a valid reason to choose another Ubuntu version than 20.04 then that is also fine, as long as a specific version is indeed selected.

**Comment by nmerriam** [comment]
Hi Rutger,

Can I ask you to add the Ubuntu version? To be honest, I just copied the
Docker file from another solution. I used WSL with Ubuntu 20.04, so I'm
sure that will be fine.

I would do it myself but I'm visiting friends and don't have access to
anything more than my smartphone and Hans is on the wrong end of a time
difference that means it's his night.

Many thanks
Nick

**Comment by rbergen** [comment]
I'm afraid that I can't do that. The correct Dockerfile needs to be part of the PR, and reviewed and merged as such. 
I think we'll just have to wait until it is possible for the Dockerfile in your PR to be updated so that it passes our CI.

**Comment by nmerriam** [comment]
Added missing ubuntu version to Dockerfile

**Comment by nmerriam** [comment]
@rbergen this looks better. Thank you for your guidance. Are you able to review and merge now?

**Comment by rbergen** [comment]
Thank you, with this I am able to review.

**Review by rbergen — APPROVED** [review]

**Review by marghidanu — APPROVED** [review]

**Pull request merged**